TFNetworkLayer.py

```diff
@@ -1892,14 +1893,18 @@ def __init__(self, activation, with_bias=True, grad_filter=None,
     :param str forward_weights_init: see :func:`TFUtil.get_initializer`
     :param str recurrent_weights_init: see :func:`TFUtil.get_initializer`
     :param str|float bias_init: see :func:`TFUtil.get_initializer`
+    :param int|None out_feature_dim_axis: if set, will convert feature dim of the input and output data
     """
     super(LinearLayer, self).__init__(**kwargs)
     from TFUtil import get_initializer
 
     self.activation = activation
     self.with_bias = with_bias
 
-    input_data = self.input_data
+    if out_feature_dim_axis is None:
+      input_data = self.input_data
+    else:
+      input_data = self.input_data.copy_with_feature_dim_axis(out_feature_dim_axis)
```

Comment:
If you specify `out_type` for a layer, you can specify the `feature_dim_axis` for the output.
Here, the code should just be like:

    input_data = self.input_data
    if not self.input_data.is_sparse:
        assert input_data.batch_ndim == self.output.batch_ndim
        input_data = input_data.copy_with_feature_dim_axis(self.output_feature_dim_axis_or_unspecified)

(Actually, did you try the code with `dot`? I'm not sure if it really works if it is channel-first. As said, please write a test case. Check that you get the same output when you don't use channel first.)